### ISSUE TITLE
AMX flag not available

### ISSUE THREAD

**Issue by Guillermogsjc** [bottlerocket-os/bottlerocket#3493]
hi,

launching m7i on EC2 through amazon linux 2 properly shows **AMX** cpu flag instructions

`lscpu | grep amx` throws amx_bf16 amx_tile amx_int8  as expected, between CPU flags:

```
fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopo
                         logy nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq monitor ssse3 fma cx16 pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hyp
                         ervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512if
                         ma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx_vnni avx512_bf16 wbnoinvd ida arat avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vae
                         s vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq rdpid cldemote movdiri movdir64b md_clear serialize amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
```

But when running a node in EKS with:
- ami: AMI Bottlerocket OS 1.15.0 (aws-k8s-1.27)
- containerd://1.6.23+bottlerocket
-  kubelet: v1.27.4-eks-cedffd4

the flags shown by lscpu, when called from a container  running at the node  with user root, are only:

```
fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon
                                     rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq monitor ssse3 fma cx16 pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_dead
                                    line_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep
                                     bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx_vnni avx51
                                    2_bf16 wbnoinvd ida arat avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq rdpid cldemote mov
                                    diri movdir64b md_clear serialize avx512_fp16 flush_l1d arch_capabilities
```

Is not expected to have the flags amx `amx_bf16  amx_tile amx_int8` accessible from Bottlerocket SO AMIs? If so, it is an incoming feature?

Kind regards.

**Comment by yeazelm** [comment]
Thanks for cutting this issue @Guillermogsjc! 

I'm assuming you launched on an Amazon Linux 2023 instance with the 6.1 kernel which has AMX support. I couldn't get this support to show up on an Amazon Linux 2 instance even with the 5.15 kernel (which makes sense). The AMX functionality landed in the 5.16 kernel which is after the 5.15 kernel that the k8s 1.27 variant uses. Bottlerocket just released a new variant for k8s 1.28 which also has this 6.1 kernel and I just confirmed it has the same enablement.

I believe it is expected that you would need this latest kernel to get support on Bottlerocket. Is it possible for you to use the 1.28 variant for this support?

From my 1.28 instance:
```
[ssm-user@control]$ lscpu | grep amx
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq monitor ssse3 fma cx16 pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx_vnni avx512_bf16 wbnoinvd ida arat avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq rdpid cldemote movdiri movdir64b md_clear serialize amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities

[ssm-user@control]$ lscpu | tr " " "\n" | grep amx
amx_bf16
amx_tile
amx_int8
```

**Comment by Guillermogsjc** [comment]
hi @yeazelm , ops, sorry for raising the issue then, in my dreams I saw kernel 6.1 linked to **AMI Bottlerocket OS 1.15.0 (aws-k8s-1.27)**, what clearly can not be the case.

thank you very much, lets go for the kernel 6.1 on the k8s 1.28 AMI
